arrays/arrays.c

```diff
@@ -19,45 +20,50 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
+Array *create_array(int capacity)
+{
   // Allocate memory for the Array struct
+  Array *array = malloc(sizeof(Array));
 
   // Set initial values for capacity and count
+  array->capacity = capacity;
+  array->count = 0;
 
   // Allocate memory for elements
-
+  array->elements = malloc(capacity * sizeof(char *));
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr)
+{
 
   // Free all elements
+  free(arr->elements);
 
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
-
+void resize_array(Array *arr)
+{
   // Create a new element storage with double capacity
+  arr->capacity *= 2;
+  arr->elements = realloc(arr->elements, arr->capacity * sizeof(char *));
```

Comment:
Nice, good use of realloc here. I've also seen this implmented using calloc() to create a new array then using a for loop to copy over each element one by one. I think this is much more concise though.